src/app/app.ts

```diff
@@ -12,18 +15,32 @@ export class App implements IApp {
   constructor(
     protected container: IContainer,
     protected router: IRouter,
-    protected errorHandler: CustomErrorHandler = DefaultErrorHandler
+    protected errorHandler: CustomErrorHandler = DefaultErrorHandler,
+    protected requestBodyDeserializer: RequestBodyDeserializer = new DefaultBodyDeserializer()
   ) {}
 
   resolve<T>(constructor: GenericConstructor<T>): T {
     return this.container.resolve(constructor)
   }
 
-  async handleRequest(request: Request): Promise<Response> {
+  async handleRequest(event: RequestEvent): Promise<Response> {
+    const [data, error] = this.getRouteData(event)
```

Comment:
why handle errors this way instead of wrapping this in a try catch block?

Reply:
dont remember 100% i think it was to basically make it so i didn't have to use `let` and/or have a large try catch body,

Reply:
Is this really worth it?

Reply:
to avoid this?
```ts
let routeData: RouteData
try {
  routeData = this.router.getRouteData(event)
} catch (e) {
  this.errorHandler(error, event)
}

Reply:
good question

Reply:
Feel free to open a PR to change